presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/HistoryTable.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.iceberg;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.prestosql.metadata.InternalTable;
+import io.prestosql.spi.connector.ColumnMetadata;
+import io.prestosql.spi.connector.ConnectorPageSource;
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.connector.ConnectorTableMetadata;
+import io.prestosql.spi.connector.ConnectorTransactionHandle;
+import io.prestosql.spi.connector.FixedPageSource;
+import io.prestosql.spi.connector.SystemTable;
+import io.prestosql.spi.predicate.TupleDomain;
+import org.apache.iceberg.HistoryEntry;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.util.SnapshotUtil;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.spi.type.BooleanType.BOOLEAN;
+import static io.prestosql.spi.type.TimestampType.TIMESTAMP;
+
+public class HistoryTable
+        implements SystemTable
+{
+    private final IcebergTableHandle tableHandle;
+    private final Table icebergTable;
+
+    private static final List<ColumnMetadata> COLUMNS = ImmutableList.of(
+            new ColumnMetadata("made_current_at", TIMESTAMP),
+            new ColumnMetadata("snapshot_id", BIGINT),
+            new ColumnMetadata("parent_id", BIGINT),
+            new ColumnMetadata("is_current_ancestor", BOOLEAN));
+
+    public HistoryTable(IcebergTableHandle tableHandle, Table icebergTable)
+    {
+        this.tableHandle = tableHandle;
+        this.icebergTable = icebergTable;
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return Distribution.SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return new ConnectorTableMetadata(tableHandle.getSchemaTableName(), COLUMNS);
```

Comment:
I had this doubt before: is it okay to use `tableHandle.getSchemaTableName()` here? Should the argument here be `test_schema.test_table` or `test_schema.test_table$history`? I think `test_schema.test_table$history` should be more accurate. But it looks like we can fix this in another PR, so that `PartitionTable` can be fixed together.

Reply:
The table name is not actually used for dynamic system tables (since the engine already had the name to ask for it). However, we should fix it to return the correct name with `$history`.